src/transport_ofi.h

```diff
@@ -206,16 +202,24 @@ typedef struct shmem_transport_ofi_bounce_buffer_t shmem_transport_ofi_bounce_bu
 
 typedef int shmem_transport_ct_t;
 
+typedef struct shmem_transport_cntr_ep_t {
+    struct fid_ep* ep;
+    struct fid_cntr*                put_cntrfd;
+    struct fid_cntr*                get_cntrfd;
+    shmem_internal_atomic_uint64_t  pending_put_counter;
+    shmem_internal_atomic_uint64_t  pending_get_counter;
+} shmem_transport_cntr_ep_t;
+
 struct shmem_transport_ctx_t {
-  struct fid_domain* domain;
-  struct fid_ep* endpoint;
-  /* etc... */
-  /* TODO fully populate this context struct */
+  shmem_transport_cntr_ep_t endpoint;
+  int id;
```

Comment:
The separation of the cntr/ctx structures seems unnecessary and may cost use extra indirections.